### PR TITLE
Implement the rest of GameEngine and Win32GameEngine

### PR DESCRIPTION
**Pull request by jonwil**

### CODE REVIEW

File: src/game/common/gameengine.cpp [hunk 1]
```diff
@@ -96,20 +101,88 @@ GameEngine::GameEngine() : m_maxFPS(0), m_isQuitting(false), m_isActive(false)
...
 
-void GameEngine::Reset() {}
+void GameEngine::Reset()
+{
+    WindowLayout *layout = g_theWindowManager->Win_Create_Layout("Menus/BlankWindow.wnd");
+    captainslog_dbgassert(layout != nullptr, "We Couldn't Load Menus/BlankWindow.wnd");
+    layout->Hide(false);
+    layout->Bring_Forward();
+    GameWindow *window = layout->Get_Window_List();
+    window->Win_Clear_Status(128);
```
Comment (tomsons26) [inline src/game/common/gameengine.cpp]: WIN_STATUS_IMAGE

File: src/game/common/gameengine.cpp [hunk 2]
```diff
@@ -96,20 +101,88 @@ GameEngine::GameEngine() : m_maxFPS(0), m_isQuitting(false), m_isActive(false)
...
+    g_theSubsystemList->Reset_All();
+
+    if (g_theGameLogic->Is_In_Multiplayer_Game()) {
```
Comment (tomsons26) [inline src/game/common/gameengine.cpp]: Is_In_Multiplayer_Game needs to be before reset all call

**Comment by tomsons26** [comment]
createThingFactory should be implemented too

comment that CComModule::Init was in GameEngine ctor

File: src/game/common/gameengine.cpp (hunk 1 above)
Comment (jonwil) [reply src/game/common/gameengine.cpp]: fixed.

File: src/game/common/gameengine.cpp (hunk 2 above)
Comment (jonwil) [reply src/game/common/gameengine.cpp]: fixed.

File: src/game/common/gamemain.cpp [hunk 3]
```diff
@@ -15,18 +15,14 @@
...
 #include "disabledtypes.h"
+#include "main.h"
 #include "win32gameengine.h"
 
 GameEngine *Create_Game_Engine()
 {
-#ifdef GAME_DLL
-    return Call_Function<GameEngine *>(PICK_ADDRESS(0x00401CD0, 0));
-#else
     GameEngine *tmp = new Win32GameEngine;
```
Comment (xezon) [inline src/game/common/gamemain.cpp]: *engine

**Review by tomsons26 — APPROVED** [review]

File: src/game/common/gameengine.cpp [hunk 4]
```diff
@@ -96,20 +101,89 @@ GameEngine::GameEngine() : m_maxFPS(0), m_isQuitting(false), m_isActive(false)
...
+    delete g_theCommandList;
+    g_theCommandList = nullptr;
+
+    delete g_theNameKeyGenerator;
+    g_theNameKeyGenerator = nullptr;
+
+    delete g_theFileSystem;
+    g_theFileSystem = nullptr;
+
+    delete g_theGameLODManager;
```
Comment (xezon) [inline src/game/common/gameengine.cpp:132]: g_theGameLODManager is not set to null

File: src/game/common/gameengine.cpp [hunk 5]
```diff
@@ -96,20 +101,89 @@ GameEngine::GameEngine() : m_maxFPS(0), m_isQuitting(false), m_isActive(false)
...
+    // un-necessary CComModule::Init call omitted
 }
 
 GameEngine::~GameEngine()
 {
-    // TODO
+    delete g_theMapCache;
+    g_theMapCache = nullptr;
+
+    delete g_theGameResultsQueue;
```
Comment (xezon) [inline src/game/common/gameengine.cpp:113]: g_theGameResultsQueue is not set to null

**Review by xezon — APPROVED** [review]

**Pull request merged**